packages/rpc-methods/src/permitted/requestSnaps.ts

```diff
@@ -124,17 +122,16 @@ async function requestSnapsImplementation(
   // TODO: Should this be part of the install flow?
 
   try {
-    // We expect the params to be the same as wallet_requestPermissions
-    const requestedPermissions = Object.keys(requestedSnaps).reduce<
-      Record<string, Partial<PermissionConstraint>>
-    >((acc, key) => {
-      acc[getSnapPermissionName(key)] = requestedSnaps[key];
-      return acc;
-    }, {});
+    const requestedPermissions = {
+      [permissionKey]: {
+        caveats: [{ type: SnapCaveatType.SnapIds, value: requestedSnaps }],
```

Comment:
At this point do we need to do this? Could we just pass the snap ids directly to the `hasSnaps` function since we are changing how that function works?

Reply:
Not sure I follow? We already pass the snap ids directly into `hasSnaps` via `requestedSnaps` and we pass `requestedPermissions` into the `requestPermissions` call.

Reply:
Ah yes, I must have misread. Makes sense 👍